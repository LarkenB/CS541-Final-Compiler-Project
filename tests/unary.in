int x = 1;
int y = -x;
char q = -'0';
float z = -x;