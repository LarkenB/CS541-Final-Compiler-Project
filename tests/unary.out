target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

define dso_local i32 @main() {
  %t1 = alloca i32
  store i32 1, i32* %t1
  %t2 = load i32, i32* %t1
  %t3 = sub i32 0, %t2
  %t4 = alloca i32
  store i32 %t3, i32* %t4
  %t5 = sub i8 0, 48
  %t6 = alloca i8
  store i8 %t5, i8* %t6
  %t7 = load i32, i32* %t1
  %t8 = sub i32 0, %t7
  %t9 = alloca float
  %t10 = sitofp i32 %t8 to float
  store float %t10, float* %t9
  ret i32 0
}
